crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -318,34 +318,20 @@ impl CumulocityConverter {
 
     pub fn process_alarm_messages(
         &mut self,
-        topic: &Topic,
-        message: &Message,
+        source: &EntityTopicId,
+        input: &Message,
+        alarm_type: &str,
     ) -> Result<Vec<Message>, ConversionError> {
-        if topic.name.starts_with("tedge/alarms") {
-            let mut mqtt_messages: Vec<Message> = Vec::new();
-            self.size_threshold.validate(message)?;
-            let mut messages = self.alarm_converter.try_convert_alarm(message)?;
-            if !messages.is_empty() {
-                // When there is some messages to be sent on behalf of a child device,
-                // this child device must be declared first, if not done yet
-                let topic_split: Vec<&str> = topic.name.split('/').collect();
-                if topic_split.len() == 5 {
-                    let child_id = topic_split[4];
-                    add_external_device_registration_message(
-                        child_id.to_string(),
-                        &mut self.children,
-                        &mut mqtt_messages,
-                    );
-                }
-            }
-            mqtt_messages.append(&mut messages);
-            Ok(mqtt_messages)
-        } else if topic.name.starts_with(INTERNAL_ALARMS_TOPIC) {
-            self.alarm_converter.process_internal_alarm(message);
-            Ok(vec![])
-        } else {
-            Err(ConversionError::UnsupportedTopic(topic.name.clone()))
-        }
+        self.size_threshold.validate(input)?;
```

Comment:
This check can be removed. Only the size of output messages must be checked.

Reply:
I tried to move this logic after creating either SmartREST or JSON over MQTT message, but one unit test failed. Therefore, I'll leave it as it is now. 

P.S. If the input JSON is way too big, most likely the output mqtt message is also too big. So, checking input is not completely useless.

Reply:
Okay, we will see later.

Reply:
> If the input JSON is way too big, most likely the output mqtt message is also too big. So, checking input is not completely useless.

Yes, that was the reason why it was done, to avoid an unnecessary conversion of a huge input that'd be thrown away a few steps later.